#Case Study on Measures of Central Tendency and Dispersion  of Students performance dataset

In [2]:
import pandas as pd
import numpy as np

In [3]:
studentdata='StudentsPerformance.csv'
df=pd.read_csv(studentdata)
df.head()

gender race/ethnicity  ... reading score writing score
0  female        group C  ...            17            10
1  female        group B  ...            24            23
2  female        group B  ...            32            28
3  female        group B  ...            38            32
4  female        group C  ...            39            33

[5 rows x 8 columns]

#1 NUMBER OF MALES AND FEMALES

In [5]:
gender = df["gender"].value_counts()
gender

female    518
male      482
Name: gender, dtype: int64

inference:518 females participated in test
          482 males participated in test 

#2 STUDENT'S PARENTAL LEVEL OF EDUCATION

In [ ]:
df['parental level of education'].value_counts().to_frame().reset 


index  parental level of education
0        some college                          226
1  associate's degree                          222
2         high school                          196
3    some high school                          179
4   bachelor's degree                          118
5     master's degree                           59

INFERENCE:1. 226 parents have  college education.
          2. out of 226, 118 have bachelors degree
             and 59 have masters degree

## 3.A STUDENTS SCORED ABOVE AVERAGE FOR MATH ,READING,WRITING BASED ON GENDER

In [ ]:
import pandas as pd
studentdata='StudentsPerformance.csv'
df=pd.read_csv(studentdata)


In [ ]:
df1 = df[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').mean()

In [ ]:
df1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

INFERENCE: Female students have higher average reading and writing score while male students have higher average in maths score..

#3.B STUDENTS SCORED ABOVE AVERAGE FOR MATH ,READING,WRITING BASED ON TEST PREPARATION COURSE

In [ ]:
df2= df[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').mean()

In [ ]:
df2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

INFERENCE: Students who did test preparation course have higher average score across all categories.

#4.A THE SCORING VARIATION FOR MATH,READING AND WRITING BASED ON GENDER

In [ ]:
genvar= df[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').std()

In [ ]:
genvar

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

INFERENCE: male students have less variations across each of the scores(14.556411 ,14.149594 ,14.227225).

#4.B THE SCORING VARIATION FOR MATH,READING AND WRITING BASED ON TEST PREPARATION COURSE

In [ ]:
testvar= df[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').std()

In [ ]:
testvar

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

Inference: The variation between those who completed and none are more or less the same.

#5)TOP 25% OF THE STUDENTS BASED ON MATH SCORE

In [ ]:
df.nlargest(250, ['math score'])

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]

Inference:List of top 250 students who are eligible for bonus points.